File: longest_common_prefix.py
from typing import List

strs = ["flower", "flow", "flight"]


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        result = ""
        for i in range(len(strs[0])):
            for j in range(1, len(strs)):
                if i >= len(strs[j]) or strs[j][i] != strs[0][i]:
                    return result
            result += strs[0][i]

        return result


result = Solution.longestCommonPrefix(Solution, strs)

print(result)

File: longest_common_prefix.php
<?php

$strs = ["flower", "flow", "flight"];

function longestCommonPrefix($strs) {
  $result = '';
  if(count($strs) == 1) return $strs[0];
  sort($strs);
  $str1 = $strs[0];
  $str2 = $strs[count($strs) - 1];
  for($i = 0; $i < strlen($str1); $i++){
      if($str2[$i] != $str1[$i]){
          return $result;
      }
      $result .=$str2[$i];
  }   
  return $result;
}

longestCommonPrefix($strs);
echo longestCommonPrefix($strs);